File: erlang/math/gcd_lcm.erl
%% Print the divisor and multiple of fixed pairs, with Euclid's algorithm written as two clauses.
%% Run: escript gcd_lcm.erl

-module(gcd_lcm).
-export([main/1]).

euclid(First, 0) ->
    First;
euclid(First, Second) ->
    euclid(Second, First rem Second).

main(_) ->
    Pairs = [{1071, 462}, {270, 192}, {17, 5}, {120, 36}],
    lists:foreach(
        fun({First, Second}) ->
            Divisor = euclid(First, Second),
            io:format("~w ~w ~w ~w~n", [First, Second, Divisor, First div Divisor * Second])
        end,
        Pairs
    ).

File: elixir/math/gcd_lcm.exs
# Print the divisor and multiple of fixed pairs, with Euclid's algorithm written as two clauses.
# Run: elixir gcd_lcm.exs

defmodule GcdLcm do
  def euclid(first, 0), do: first
  def euclid(first, second), do: euclid(second, rem(first, second))
end

[{1071, 462}, {270, 192}, {17, 5}, {120, 36}]
|> Enum.each(fn {first, second} ->
  divisor = GcdLcm.euclid(first, second)
  IO.puts("#{first} #{second} #{divisor} #{div(first, divisor) * second}")
end)
